frontend/lib/framework/UniqueString.cpp

```diff
@@ -184,6 +184,25 @@ UniqueString UniqueString::get(Context* context,
   }
 }
 
+int UniqueString::compare(const UniqueString other) const {
+  if (*this == other) return 0;
+  size_t lenThis = this->length();
+  size_t lenOther = other.length();
+  size_t minsize = (lenThis < lenOther) ? lenThis : lenOther;
+  int cmp = memcmp(this->c_str(), other.c_str(), minsize);
+  if (cmp == 0) {
+    if (lenThis == lenOther) return 0;
+    return (lenThis < lenOther) ? -1 : 1;
+  }
+  return (cmp < 0) ? -1 : 1;
+}
+
+int UniqueString::compare(const char* other) const {
+  int cmp = strcmp(this->c_str(), other);
+  if (cmp == 0) return 0;
+  return (cmp < 0) ? -1 : 1;
+}
+
```

Comment:
Oh, is it because you want to go through null terminators? If so, why do you need this?

Reply:
This and the previous function were documented to return `-1` `0` or `1` and I wanted to make an improvement (enabling comparison of UniqueStrings containing interior null bytes) without changing this aspect of the API.

I suspect that it does not matter in practice and we could make `UniqueString::compare` return "less than zero" "zero" "greater than zero" and it would be fine. I just did not want to get in to it here.